app/models/repository.rb

```diff
@@ -1,3 +1,7 @@
 class Repository < KalibroClient::Entities::Processor::Repository
   include KalibroRecord
+
+  def self.latest(count=1)
+    all.sort { |one, another| another.id <=> one.id }.first(count)
```

Comment:
What I wrote was meant to address all of the questions, which ones in particular do you mean? I don't think the cache justifies writing code that is two or three orders of magnitude less optimal than what is possible (and without that much effort).